joblib/test/test_numpy_pickle.py

```diff
@@ -290,38 +290,33 @@ def test_compress_mmap_mode_warning(tmpdir):
     a = rnd.random_sample(10)
     this_filename = tmpdir.join('test.pkl').strpath
     numpy_pickle.dump(a, this_filename, compress=1)
-    with warnings.catch_warnings(record=True) as caught_warnings:
-        warnings.simplefilter("always")
+    with warns(UserWarning) as warninfo:
         numpy_pickle.load(this_filename, mmap_mode='r+')
-        assert len(caught_warnings) == 1
-        for warn in caught_warnings:
-            assert warn.category == UserWarning
-            assert (warn.message.args[0] ==
-                    'mmap_mode "%(mmap_mode)s" is not compatible with '
-                    'compressed file %(filename)s. "%(mmap_mode)s" flag will '
-                    'be ignored.' % {'filename': this_filename,
-                                     'mmap_mode': 'r+'})
+    assert len(warninfo) == 1
+    assert (str(warninfo[-1].message) ==
+            'mmap_mode "%(mmap_mode)s" is not compatible with compressed '
+            'file %(filename)s. "%(mmap_mode)s" flag will be ignored.' %
+            {'filename': this_filename, 'mmap_mode': 'r+'})
 
 
 @with_numpy
-def test_cache_size_warning(tmpdir):
+def test_cache_size_warning(tmpdir, recwarn):
     # Check deprecation warning raised when cache size is not None
     filename = tmpdir.join('test.pkl').strpath
     rnd = np.random.RandomState(0)
     a = rnd.random_sample((10, 2))
 
     for cache_size in (None, 0, 10):
-        with warnings.catch_warnings(record=True) as caught_warnings:
-            warnings.simplefilter("always")
-            numpy_pickle.dump(a, filename, cache_size=cache_size)
-            expected_nb_warnings = 1 if cache_size is not None else 0
-            assert len(caught_warnings) == expected_nb_warnings
-            for warn in caught_warnings:
-                assert warn.category == DeprecationWarning
-                assert (warn.message.args[0] ==
-                        "Please do not set 'cache_size' in joblib.dump, this "
-                        "parameter has no effect and will be removed. You "
-                        "used 'cache_size={0}'".format(cache_size))
+        warnings.simplefilter("always")
+        numpy_pickle.dump(a, filename, cache_size=cache_size)
+        expected_nb_warnings = 1 if cache_size is not None else 0
```

Comment:
You can use `with warns(None) as record` as the doc says for this use case instead of using `recwarn`